ndb/src/google/cloud/ndb/key.py

```diff
@@ -705,26 +716,37 @@ def get(self, **ctx_options):
                 For example, ``{"read_policy": EVENTUAL_CONSISTENCY}``.
 
         Raises:
-            NotImplementedError: Always. The method has not yet been
-                implemented.
+            NotImplementedError: When ``ctx_options`` is not empty. Context
+                options have not yet been implemented.
+
+        Returns:
+            Union[:class:`.Model`, :data:`None`]
         """
-        raise NotImplementedError
+        return self.get_async(**ctx_options).result()
 
+    @tasklets.tasklet
     def get_async(self, **ctx_options):
         """Asynchronously get the entity for this key.
 
-        The result for the returned future with either by the retrieved
+        The result for the returned future will either be the retrieved
         :class:`.Model` or :data:`None` if there is no such entity.
 
         Args:
             ctx_options (Dict[str, Any]): The context options for the request.
                 For example, ``{"read_policy": EVENTUAL_CONSISTENCY}``.
 
         Raises:
-            NotImplementedError: Always. The method has not yet been
-                implemented.
+            NotImplementedError: When ``ctx_options`` is not empty. Context
+                options have not yet been implemented.
+
+        Returns:
+            :class:`~google.cloud.ndb.tasklets.Future`
         """
-        raise NotImplementedError
+        from google.cloud.ndb import model  # avoid circular import
```

Comment:
I'm guessing that because the module names and their relationships are basically defined by the previous version, it's very challenging to restructure to avoid circular imports in the first place, right?

Reply:
Precisely. I think if we had carte blanche to refactor everything, I'd argue that everything that could result in an API call (get, put, delete, query) should be on Client. So this would be `Client.get_async(key)`.